### PR TITLE
Trace library (GAPIC-only)

### PR DESCRIPTION
**Pull request by lukesneeringer**
/cc @duggelz

### CODE REVIEW

File: trace/setup.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        'Intended Audience :: Developers',
+        'Development Status :: 3 - Alpha',
+        'Intended Audience :: Developers',
+        'License :: OSI Approved :: Apache Software License',
+        'Programming Language :: Python',
+        'Programming Language :: Python :: 2',
+        'Programming Language :: Python :: 2.7',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.4',
+        'Programming Language :: Python :: 3.5',
```
Comment (duggelz) [inline trace/setup.py]: Python 3.6?

File: trace/setup.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import sys
+
+install_requires = [
+    'google-gax>=0.15.7, <0.16dev',
+    'googleapis-common-protos[grpc]>=1.5.2, <2.0dev',
+]
+
+setup(
+    name='google-cloud-trace',
+    version='0.15.4',
```
Comment (duggelz) [inline trace/setup.py]: I'm curious what 0.15.4 corresponds to?

File: trace/PUBLISHING.rst [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+Note: This folder has been generated by the GAPIC code generator.
+
+The instructions assumes that no changes have been made to folder and its
+contents since it was created.
+
+PREREQUISITES
+-------------
+
+- Python must installed
+- [tox](https://testrun.org/tox/latest/) must be installed
```
Comment (duggelz) [inline trace/PUBLISHING.rst]: nox?

File: trace/google/cloud/gapic/trace/v1/enums.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# Copyright 2016 Google Inc. All rights reserved.
```
Comment (duggelz) [inline trace/google/cloud/gapic/trace/v1/enums.py]: 2017

File: trace/google/cloud/gapic/trace/v1/trace_service_client.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+                 scopes=None,
+                 client_config=None,
+                 app_name=None,
+                 app_version='',
+                 lib_name=None,
+                 lib_version='',
+                 metrics_headers=()):
+        """Constructor.
+
+        Args:
```
Comment (duggelz) [inline trace/google/cloud/gapic/trace/v1/trace_service_client.py]: Is this the right documentation style for this repository?  This file doesn't quite match the style in enums.py

Comment (dhermes) [inline trace/google/cloud/gapic/trace/v1/trace_service_client.py]: I am +1000 :+1: on switching to Google-style (via Napolean) docstrings

File: trace/setup.py (hunk 1 above)
Comment (lukesneeringer) [inline trace/setup.py]: Need to fix this in autogen.

File: trace/PUBLISHING.rst (hunk 3 above)
Comment (lukesneeringer) [inline trace/PUBLISHING.rst]: I manually added a `nox.py`; autogen still uses `tox`. Need to fix this.

File: trace/setup.py (hunk 2 above)
Comment (lukesneeringer) [inline trace/setup.py]: All the GAPICs more or less follow the same versioning scheme right now, and 0.15.4 is where we are.

We now (as of last month) have the ability to override this on a per-API basis, so maybe we should start this at 0.1.0?

**Comment by duggelz** [comment]
Luke, hold off on merging this, it looks like some changes to the API and more are coming down the pike.

**Comment by lukesneeringer** [comment]
> Luke, hold off on merging this, it looks like some changes to the API and more are coming down the pike.

Copy that, thanks. Closing for now, since it will likely be a newly-generated library when the time comes.

**Pull request closed without merging**